rancid-tomatillos/src/App.js

```diff
@@ -1,30 +1,54 @@
 import React, {Component} from "react";
+import {BrowserRouter as Router, Switch, Route, Link} from "react-router-dom";
 import MainView from "./components/MainView";
-import CardsContainer from "./components/CardsContainer";
+import Header from "./components/Header";
+import Login from "./components/Login";
+import ShowPage from "./components/ShowPage";
 
 import "./App.css";
 
 class App extends Component {
     constructor() {
         super();
+        this.setLoggedIn = this.setLoggedIn.bind(this);
         this.state = {
-            movies: null
+            movies: null,
+            userId: 0,
+            loggedIn: false,
         };
     }
-    componentDidMount() {
-        return fetch("https://rancid-tomatillos.herokuapp.com/api/v2/movies")
-            .then((response) => response.json())
-            .then((data) => {
-                this.setState({movies: data.movies});
-            });
+    async componentDidMount() {
+        let promise = await fetch(
+            "https://rancid-tomatillos.herokuapp.com/api/v2/movies"
+        );
+        let result = await promise.json();
+        this.setState({movies: result.movies});
+    }
+    setLoggedIn() {
+        this.setState({loggedIn: true});
```

Comment:
Will probably need to make this dynamic for log out functionalty

Reply:
Yes, and add UI for errors